From 2329bda157d5c9ebb45903ebd713fe8bd7e2a5c6 Mon Sep 17 00:00:00 2001
From: Zhantao Tang <zhantao.tang@windriver.com>
Date: Fri, 15 Jul 2022 15:57:41 +0800
Subject: [PATCH] pfeng: fix to compatible with kernel v5.15

This patch updates pfeng slave driver codes to compatible with
linux kernel v5.15, to fix build errors.

Upstream-Status: Pending

Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 sw/linux-pfeng/pfeng-dt.c      |  4 ++--
 sw/linux-pfeng/pfeng-ethtool.c | 12 +++++++++---
 2 files changed, 11 insertions(+), 5 deletions(-)

diff --git a/sw/linux-pfeng/pfeng-dt.c b/sw/linux-pfeng/pfeng-dt.c
index aa1343f..e19bdd6 100644
--- a/sw/linux-pfeng/pfeng-dt.c
+++ b/sw/linux-pfeng/pfeng-dt.c
@@ -274,8 +274,8 @@ int pfeng_dt_create_config(struct pfeng_priv *priv)
 		dev_info(dev, "netif name: %s", netif_cfg->name);
 
 		/* MAC eth address */
-		netif_cfg->macaddr = (u8 *)of_get_mac_address(child);
-		if (netif_cfg->macaddr)
+		ret = of_get_mac_address(child, netif_cfg->macaddr);
+		if (!ret)
 			dev_info(dev, "DT mac addr: %pM", netif_cfg->macaddr);
 
 		if (of_find_property(child, "nxp,pfeng-netif-mode-aux", NULL))
diff --git a/sw/linux-pfeng/pfeng-ethtool.c b/sw/linux-pfeng/pfeng-ethtool.c
index 6e7334d..e937527 100644
--- a/sw/linux-pfeng/pfeng-ethtool.c
+++ b/sw/linux-pfeng/pfeng-ethtool.c
@@ -114,7 +114,10 @@ static int pfeng_ethtool_get_ts_info(struct net_device *netdev, struct ethtool_t
 	return 0;
 }
 
-static int pfeng_get_coalesce(struct net_device *netdev, struct ethtool_coalesce *ec)
+static int pfeng_get_coalesce(struct net_device *netdev,
+                              struct ethtool_coalesce *ec,
+                              struct kernel_ethtool_coalesce *kernel_coal,
+                              struct netlink_ext_ack *extack)
 {
 	struct pfeng_netif *netif = netdev_priv(netdev);
 	struct pfeng_hif_chnl *chnl;
@@ -134,7 +137,10 @@ static int pfeng_get_coalesce(struct net_device *netdev, struct ethtool_coalesce
 	return 0;
 }
 
-static int pfeng_set_coalesce(struct net_device *netdev, struct ethtool_coalesce *ec)
+static int pfeng_set_coalesce(struct net_device *netdev,
+                              struct ethtool_coalesce *ec,
+                              struct kernel_ethtool_coalesce *kernel_coal,
+                              struct netlink_ext_ack *extack)
 {
 	struct pfeng_netif *netif = netdev_priv(netdev);
 	struct pfeng_hif_chnl *chnl;
@@ -229,7 +235,7 @@ int pfeng_ethtool_params_restore(struct pfeng_netif *netif) {
 	ec.rx_max_coalesced_frames = chnl->cfg_rx_max_coalesced_frames;
 	ec.rx_coalesce_usecs = chnl->cfg_rx_coalesce_usecs;
 
-	ret = pfeng_set_coalesce(netdev, &ec);
+	ret = pfeng_set_coalesce(netdev, &ec, NULL, NULL);
 	if (ret)
 		netdev_warn(netdev, "Coalescing not restored\n");
 
-- 
2.25.1

